lib/components/user/places/place-editor.tsx

```diff
@@ -67,18 +83,51 @@ function makeLocationFieldLocation(favoriteLocation: UserSavedLocation) {
 class PlaceEditor extends Component<Props> {
   static contextType = ComponentContext
 
-  _handleLocationChange = (e: LocationSelectedEvent) => {
-    const { setTouched, setValues, values } = this.props
-    const { lat, lon, name } = e.location
+  _setLocation = (location: Location) => {
+    const { intl, setTouched, setValues, values } = this.props
+    const { category, lat, lon, name } = location
     setValues({
       ...values,
-      address: name,
+      address:
+        // If the raw current location is passed without a name attribute (i.e. the address),
+        // set the "address" as the formatted coordinates of the current location at that time.
+        category === 'CURRENT_LOCATION'
+          ? intl.formatMessage({ id: 'common.coordinates' }, { lat, lon })
+          : name,
       lat,
       lon
     })
     setTouched({ address: true })
   }
 
+  _handleLocationChange = (e: LocationSelectedEvent) => {
+    this._setLocation(e.location)
+  }
+
+  _handleGetCurrentPosition = () => {
+    const { geocoderConfig, getCurrentPosition, intl } = this.props
+    getCurrentPosition(
+      intl,
+      locationActions.PLACE_EDITOR_LOCATION,
+      ({ coords }) => {
+        const { latitude: lat, longitude: lon } = coords
+        // Populate the "address" field with the coordinates at first.
+        // If geocoding succeeds, the resulting address will appear there.
+        this._setLocation({
+          category: 'CURRENT_LOCATION',
+          lat,
+          lon
+        })
+        getGeocoder(geocoderConfig)
+          .reverse({ point: coords })
+          .then(this._setLocation)
+          .catch((err: Error) => {
+            console.warn(err)
```

Comment:
Will this prevent bugsnag from detecting errors resulting from failed geocoding requests? Do we care?

Reply:
Do you have an example in mind you can point me to?

Reply:
I think I copied some other geocoder code that does the same thing.